main.cpp

```diff
@@ -174,21 +156,8 @@ struct IntType
     IntType& subtract( int rhs );
     IntType& multiply( int rhs );
     IntType& divide( int rhs );
-    // 4 functions taking a FloatType as input
-    IntType& add(const FloatType& rhsFt );
-    IntType& subtract(const FloatType& rhsFt );
-    IntType& multiply(const FloatType& rhsFt );
-    IntType& divide(const FloatType& rhsFt );
-    // 4 functions taking a DoubleType as input
-    IntType& add(const DoubleType& rhsDt );
-    IntType& subtract(const DoubleType& rhsDt );
-    IntType& multiply(const DoubleType& rhsDt );
-    IntType& divide(const DoubleType& rhsDt );
-    // 4 functions taking an int as input
-    IntType& add(const IntType& rhsIt );
-    IntType& subtract(const IntType& rhsIt );
-    IntType& multiply(const IntType& rhsIt );
-    IntType& divide(const IntType& rhsIt );    
+    // convert UDT to primitive type
+    operator int();
```

Comment:
It's easiest to just implement these in-class
- [ ] move conversion function implementation to inside of class.